pkg/api/defaults.go

```diff
@@ -644,6 +647,9 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 		if p.IsAzureStackCloud() && profile.PlatformFaultDomainCount == nil {
 			profile.PlatformFaultDomainCount = to.IntPtr(DefaultAzureStackFaultDomainCount)
 		}
+		if profile.PlatformUpdateDomainCount == nil {
```

Comment:
@serbrech FYI, we are making PlatformUpdateDomainCount an api model property, and the default setting of `3` is being moved into defaults enforcement (prior, it was statically set in the ARM template). Thus, AKS will have to incorporate this defaults change, otherwise it will generate ARM templates with no PlatformUpdateDomainCount configuration, which will probably fail.